apis/bigqueryanalyticshub/v1alpha1/bigqueryanalyticshubdataexchange_types.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var BigQueryAnalyticsHubDataExchangeGVK = GroupVersion.WithKind("BigQueryAnalyticsHubDataExchange")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BigQueryAnalyticsHubDataExchangeSpec defines the desired state of BigQueryAnalyticsHubDataExchange
+// +kcc:proto=google.cloud.bigquery.analyticshub.v1.DataExchange
+type BigQueryAnalyticsHubDataExchangeSpec struct {
+	// Output only. The resource name of the data exchange.
+	//  e.g. `projects/myproject/locations/US/dataExchanges/123`.
+	Name *string `json:"name,omitempty"`
+
+	// Required. Human-readable display name of the data exchange. The display
+	//  name must contain only Unicode letters, numbers (0-9), underscores (_),
+	//  dashes (-), spaces ( ), ampersands (&) and must not start or end with
+	//  spaces. Default value is an empty string. Max length: 63 bytes.
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Optional. Description of the data exchange. The description must not
+	//  contain Unicode non-characters as well as C0 and C1 control codes except
+	//  tabs (HT), new lines (LF), carriage returns (CR), and page breaks (FF).
+	//  Default value is an empty string.
+	//  Max length: 2000 bytes.
+	Description *string `json:"description,omitempty"`
+
+	// Optional. Email or URL of the primary point of contact of the data
+	//  exchange. Max Length: 1000 bytes.
+	PrimaryContact *string `json:"primaryContact,omitempty"`
+
+	// Optional. Documentation describing the data exchange.
+	Documentation *string `json:"documentation,omitempty"`
+
+	// Optional. Base64 encoded image representing the data exchange. Max
+	//  Size: 3.0MiB Expected image dimensions are 512x512 pixels, however the API
+	//  only performs validation on size of the encoded data. Note: For byte
+	//  fields, the content of the fields are base64-encoded (which increases the
+	//  size of the data by 33-36%) when using JSON on the wire.
+	Icon *string `json:"icon,omitempty"`
```

Comment:
I'm wondering if we should NOTYET this.  The idea of 3MB of encoded image data scares me :-)